packages/ec-core/src/generator/wrappers/owned.rs

```diff
@@ -0,0 +1,57 @@
+use std::{borrow::Borrow, marker::PhantomData, num::NonZeroUsize};
+
+use rand::{distributions::Uniform, prelude::Distribution};
+
+use crate::generator::wrappers::slice_cloning::EmptySlice;
+
+#[derive(Debug, PartialEq)]
+pub struct OneOfCloning<T, U> {
+    collection: T,
+    range: Uniform<usize>,
+    _p: PhantomData<U>,
+}
+
+impl<T, U> OneOfCloning<T, U>
+where
+    T: Borrow<[U]>,
+{
+    /// Create a new [`OneOfCloning`] distribution, which selects a
+    /// value from a collection and then returns a new value by cloning the
+    /// selected value.
+    ///
+    /// # Errors
+    /// - [`EmptySlice`] if an empty collection is passed in, since then no
+    ///   element can be selected from there
+    pub fn new(val: T) -> Result<Self, EmptySlice> {
+        match val.borrow().len() {
+            0 => Err(EmptySlice),
+            len => Ok(Self {
+                collection: val,
+                range: Uniform::new(0, len),
+                _p: PhantomData,
+            }),
+        }
+    }
+
+    pub fn choices(&self) -> NonZeroUsize {
```

Comment:
`choices()` seems an odd name to me. I would have thought it would give me some collection of options (choices), but it returns a number. Should it instead be `num_choices()`? Or just `len()`?

Reply:
I got the same comment in https://github.com/rust-random/rand/pull/1402 and I changed it to num_choices there so let's do that here as well.